library/script-library/src/main/java/uk/gov/gchq/gaffer/script/operation/platform/LocalDockerPlatform.java

```diff
@@ -0,0 +1,169 @@
+        }
+
+        return dockerImage;
+    }
+
+    /**
+     * Builds a docker image and creates a docker container instance.
+     *
+     * @param image                  the image to create a container from
+     * @param ip                     the ip the container is connected to
+     * @return the docker container
+     */
+    @Override
+    public Container createContainer(final Image image, final String ip) {
+
+        String containerId = "";
+        // Keep trying to create a container and find a free port.
+        try {
+            port = RandomPortGenerator.getInstance().generatePort();
+
+            // Create a container from the image and bind ports
+            final ContainerConfig containerConfig = ContainerConfig.builder().hostConfig(HostConfig.builder().portBindings(ImmutableMap.of("80/tcp", Collections.singletonList(PortBinding.of(ip, port)))).build()).image(image.getImageString()).exposedPorts("80/tcp").cmd("sh", "-c", "while :; do sleep 1; done").build();
+            final ContainerCreation creation = docker.createContainer(containerConfig);
+            containerId = creation.id();
+        } catch (final DockerException | InterruptedException e) {
+            e.printStackTrace();
+        }
+        return new LocalDockerContainer(containerId, port);
+    }
+
+    /**
+     * Starts a docker container
+     *
+     * @param container             the container
+     */
+    private void startContainer(final Container container) {
+        for (int i = 0; i < 100; i++) {
+            try {
+                LOGGER.info("Starting the Docker container...");
+                docker.startContainer(container.getContainerId());
+                break;
+            } catch (final DockerException | InterruptedException ignored) {
+            }
+        }
+    }
+
+    /**
+     * Stops and closes a docker container
+     *
+     * @param container             the container
+     */
+    private void closeContainer(final Container container) {
+        try {
+            LOGGER.info("Closing the Docker container...");
+            docker.waitContainer(container.getContainerId());
+            docker.removeContainer(container.getContainerId());
+        } catch (final DockerException | InterruptedException e) {
+            e.printStackTrace();
+            LOGGER.info("Failed to stop the container");
+        }
+    }
+
+    /**
+     * Runs a docker container
+     *
+     * @param container              the container to run
+     * @param inputData              the data to pass to the container
+     * @return the result of the container
+     */
+    @Override
+    public StringBuilder runContainer(final Container container, final Iterable inputData) {
+        startContainer(container);
+        container.sendData(inputData);
+        StringBuilder output = container.receiveData();
+        closeContainer(container);
+        RandomPortGenerator.getInstance().freePort(port);
```

Comment:
Consider creating this instance in the constructor and then using the local reference.